src/broadcast_socket.cpp

```diff
@@ -56,10 +56,10 @@ using namespace std::placeholders;
 
 namespace libtorrent
 {
-	bool is_ip_address(char const* host)
+	bool is_ip_address(string_view host)
 	{
 		error_code ec;
-		address::from_string(host, ec);
+		address::from_string(host.data(), ec);
```

Comment:
what's returned from ``data()`` is not necessarily 0-terminated, so I don't think this is right